orte/mca/rml/ofi/rml_ofi_component.c

```diff
@@ -0,0 +1,1091 @@
+            if ( (wc.flags & FI_SEND) || ((wc.flags & FI_SEND) && (wc.flags & FI_TRANSMIT_COMPLETE)) )
+            {
+                opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                       "%s Send completion received on conduitid %d",
+                       ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),
+                       conduit->conduit_id);
+                if (NULL != wc.op_context) {
+                    /* get the context from the wc and call the message handler */
+                    ofi_req = TO_OFI_REQ(wc.op_context);
+                    assert(ofi_req);
+                    ret = orte_rml_ofi_send_callback(&wc, ofi_req);
+                    if (ORTE_SUCCESS != ret) {
+                        opal_output(orte_rml_base_framework.framework_output,
+                        "Error returned by OFI request event callback: %zd",
+                         ret);
+                         abort();
+                    }
+                    
+                }
+            } else if ( (wc.flags & FI_RECV) && (wc.flags & FI_MULTI_RECV) ) {
+                    opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                         "%s Received message on conduitid %d - but buffer is consumed, need to repost",
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),
+                         conduit->conduit_id);
+                         // reposting buffer
+                         ret = fi_recv(orte_rml_ofi.ofi_conduits[conduit->conduit_id].ep,
+                          orte_rml_ofi.ofi_conduits[conduit->conduit_id].rxbuf,
+                          orte_rml_ofi.ofi_conduits[conduit->conduit_id].rxbuf_size,
+                          fi_mr_desc(orte_rml_ofi.ofi_conduits[conduit->conduit_id].mr_multi_recv),
+                          0,&(conduit->rx_ctx1)); 
+                        // call the receive message handler that will call the rml_base
+                        ret = orte_rml_ofi_recv_handler(&wc, conduit->conduit_id);
+
+            } else if ( wc.flags & FI_RECV )  {
+                    opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                         "%s Received message on conduitid %d",
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),
+                         conduit->conduit_id);
+                        // call the receive message handler that will call the rml_base
+                        ret = orte_rml_ofi_recv_handler(&wc, conduit->conduit_id);
+            } else if ( wc.flags & FI_MULTI_RECV ) {
+                    opal_output_verbose(1, orte_rml_base_framework.framework_output,
+                         "%s Received buffer overrun message on conduitid %d - need to repost",
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),
+                         conduit->conduit_id);
+                        // reposting buffer
+                        ret = fi_recv(orte_rml_ofi.ofi_conduits[conduit->conduit_id].ep,
```

Comment:
This `ret` value (and several `ret` values from `if` blocks above this) is never checked.

There's also a lot of bad indenting in this file.